src/platform/packages/shared/kbn-unified-tabs/src/components/tabs_bar/tabs_bar.tsx

```diff
@@ -121,6 +133,15 @@ export const TabsBar = forwardRef<TabsBarApi, TabsBarProps>(
         tabsContainerElement,
       });
 
+    useEffect(() => {
+      if (hasReachedMaxItemsCount && !hasSentMaxReachedEventRef.current) {
+        hasSentMaxReachedEventRef.current = true;
+        onEvent('tabsLimitReached');
+      } else if (!hasReachedMaxItemsCount) {
+        hasSentMaxReachedEventRef.current = false;
+      }
+    }, [hasReachedMaxItemsCount, onEvent]);
```

Comment:
Nit: I might be missing something, but seems like instead of juggling `hasSentMaxReachedEventRef` you could just check `hasReachedMaxItemsCount` and keep a stable ref to the event handler:
```ts
const onTabsLimitReached = useLatest(() => onEvent('tabsLimitReached'));

useEffect(() => {
  if (hasReachedMaxItemsCount) {
    onTabsLimitReached.current();
  }
}, [hasReachedMaxItemsCount]);
```

Reply:
Ha! `useLatest` is something I've never heard of, TIL, thanks!

Reply:
Adjusted here:[729192c](https://github.com/elastic/kibana/pull/234344/commits/729192c68ee7d10f61230b0bc661f56f7a575aad)
Works like a charm, thanks!